### ISSUE TITLE
feature weights

### ISSUE THREAD

**Issue by david-waterworth** [catboost/catboost#611]
Is there any way of weighting features, i.e. in order to influence which feature is chooses as the "best" to split on?

My use case is as follows. I'm trying to predict electricity demand from climate and calendar variables, i.e. temperature, month (12 string levels), dayofweek (7 string levels), hourofday (24 integer levels). (I have a lot more features but they're the primary ones)

There's strong social/economic drivers of demand as a function of dayofweek  and timeofday, and temperature is a predominant influence, month is really just a proxy for unmeasured / complex climate interactions - it's highly correlated with temperature.

My issue is I only have around 1 years data to train with and 1 to cross validate. So adding month results in overfitting, but catboost puts a very higher degree of importance on it than temperature. So my options are to leave it out (which almost doubles the out of sample r^2 from 0.50 to 0.80), or use some sort of lower dimensional approximation (i.e. a single sin and cos term, or add a feature like sunshine hours per day).

Is there any way of assigning a weight to the monthly feature, so it's used only after temperature has explained as much variance as possible?

Another option would be if at split the "best" feature is selected using some sort of cross validation. I'd imagine this would be horrible slow and unnecessary in most cases though.

**Comment by annaveronika** [comment]
I'm looking through the old issues - very sorry for not answering!

We did experiments with feature weighting, and it never helped, so we decided to not add this to catboost. As for month, our experience when working with dates shows that sin and cos work really well, so I would try going with them instead of adding categorical month.

I'm closing this one as won't fix, because we don't have a plan to implement this feature...

Than you very much for your activity in this repo!
